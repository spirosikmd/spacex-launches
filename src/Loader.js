import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  root: {
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
};

const Loader = ({ classes }) => {
  return (
    <div className={classes.root}>
      <CircularProgress />
    </div>
  );
};

Loader.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(React.memo(Loader));
