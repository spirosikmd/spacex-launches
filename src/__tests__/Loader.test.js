import Loader from '../Loader';

describe('Loader', () => {
  let props;

  beforeEach(() => {
    props = {
      classes: {
        root: 'root',
      },
    };
  });

  it('renders a loader', () => {
    expect(mountComponent(Loader, props)).toMatchSnapshot();
  });
});
