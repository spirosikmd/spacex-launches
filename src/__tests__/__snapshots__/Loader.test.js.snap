// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders a loader 1`] = `
<div
  className="root"
>
  <WithStyles(CircularProgress) />
</div>
`;
